Response/JWTAuthenticationFailureResponse.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace Lexik\Bundle\JWTAuthenticationBundle\Response;
+
+use Symfony\Component\HttpFoundation\JsonResponse;
+
+/**
+ * JWTAuthenticationFailureResponse.
+ *
+ * Response sent on failed JWT authentication (can be replaced by a custom Response).
+ *
+ * @internal
+ *
+ * @author Robin Chalas <robin.chalas@gmail.com>
+ */
+final class JWTAuthenticationFailureResponse extends JsonResponse
+{
+    /**
+     * The response message.
+     *
+     * @var string
+     */
+    private $message;
+
+    /**
+     * @param string $message A failure message passed in the response body
+     */
+    public function __construct($message = 'Bad credentials', $statusCode = JsonResponse::HTTP_UNAUTHORIZED)
+    {
+        $this->message = $message;
+
+        parent::__construct(null, $statusCode, ['WWW-Authenticate' => 'Bearer']);
+
+        $this->setData([
+            'code'    => $statusCode,
+            'message' => $message,
+        ]);
+    }
+
+    /**
+     * Sets the failure message.
+     *
+     * @param string $message
+     *
+     * @return JWTAuthenticationFailureResponse
+     */
+    public function setMessage($message)
+    {
+        $this->message = $message;
```

Comment:
Is it expected that the JSON body keep the initial message when the message is modified ?